src/byrow.jl

```diff
@@ -52,13 +61,16 @@ function byrow_helper(df, body, deprecation_warning)
     # byrow itself because then it will be displayed when the macro is evaluated.
     deprecation_warning && @warn "`@byrow!` is deprecated, use `@byrow` instead."
     e_body, e_newcols = byrow_find_newcols(body, Any[])
+    _df = gensym()
     quote
-        _N = length($df[!, 1])
-        _DF = @transform($df, $(e_newcols...))
-        $(with_helper(:_DF, :(for row = 1:_N
-            $(byrow_replace(e_body))
-        end)))
-        _DF
+        let $_df = $df
+            local _N = nrow($_df)
```

Comment:
what happens if you write to `_N` or `_DF`?

Reply:
Nothing bad. Since `_N` and `_DF` are only used when constructing new columns, something that happens before the anonymous function is evaluated, you don't break anything. 

You just don't get to use `_N` and `_DF` if you re-define them. 

Same with `row`. If the user has the correct mental model, that you are really writing `df.x[row]` then it should be obvious why things break.